### PR TITLE
feat: binance connect error handling

### PR DESCRIPTION
**Pull request by AryanCypherock**

### CODE REVIEW

File: packages/cysync-core/src/utils/error/index.ts [hunk 1]
```diff
@@ -27,6 +30,14 @@ export * from './types';
  */
 const identifyServerErrors = (error: any) => {
   if (error?.isAxiosError) {
+    if (error.response?.data?.code) {
```
Comment (TejasvOnly) [inline packages/cysync-core/src/utils/error/index.ts:33]: Is this condition sufficient? can't any other api provide a response with similar structure?

File: packages/cysync-core/src/context/buySell/index.tsx [hunk 2]
```diff
@@ -194,44 +195,54 @@ export const BuySellProvider: FC<BuySellContextProviderProps> = ({
     setUnhandledError(e);
   }, []);
 
+  const onRetry = useCallback(() => {
+    setState(BuySellState.CURRENCY_SELECT);
+    resetUserInput();
+  }, []);
```
Comment (TejasvOnly) [inline packages/cysync-core/src/context/buySell/index.tsx:201]: Are all errors supposed to retry the flow from the start?

File: packages/cysync-core/src/context/buySell/index.tsx [hunk 3]
```diff
@@ -194,44 +195,54 @@ export const BuySellProvider: FC<BuySellContextProviderProps> = ({
...
+        },
+      );
 
-    setCryptoDropdownList(cryptoDropdown);
-    return true;
+      setCryptoDropdownList(cryptoDropdown);
+      return true;
+    } catch (err) {
+      onError(err);
+      return false;
```
Comment (TejasvOnly) [inline packages/cysync-core/src/context/buySell/index.tsx:244]: All errors are piped directly to `unhandledError`, have you considered what will happen if the error is not a server error?

**Review by TejasvOnly — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/utils/error/index.ts (hunk 1 above)
Comment (AryanCypherock) [reply packages/cysync-core/src/utils/error/index.ts:33]: Will pick in a different PR

File: packages/cysync-core/src/context/buySell/index.tsx (hunk 2 above)
Comment (AryanCypherock) [reply packages/cysync-core/src/context/buySell/index.tsx:201]: Will pick in a different PR

File: packages/cysync-core/src/context/buySell/index.tsx (hunk 3 above)
Comment (AryanCypherock) [reply packages/cysync-core/src/context/buySell/index.tsx:244]: Will pick in a different PR

**Pull request merged**